template/logEntry/logEntry.proto

```diff
@@ -0,0 +1,94 @@
+// Copyright 2017 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+
+package logEntry;
+
+// TODO: STUFF THAT IS COMMENTED OUT NEEDS TO BE TURNED ON ONCE CODE GEN IS FIXED
```

Comment:
I'm not seeing things the same way at this point.

Think of a template as a generic type definition: Foo<T>. When you define Foo<T>, certain fields can have definite types, like string or int, while other fields are of type T. You can then define a type that derives from Foo<T> and defines what T means.

In the same way, fields in templates can be ValueType, indicating these fields' types must be defined by a concrete type. Fields of other types are already fully specified and so are used directly in the Type. So the fact the template has a field of type Severity means the Type will have an identical field, and thus the instance YAML will need to supply a value of type Severity. So at the point where a log entry is created, the severity is assigned on the spot. I think the same thing is true for timestamp and log name.

I agree that the payload format and template stuff should go elsewhere.